core/src/banking_stage.rs

```diff
@@ -4691,53 +4247,205 @@ mod tests {
+            new_vote_state_update_transaction(
+                VoteStateUpdate::default(),
+                Hash::new_unique(),
+                &keypair,
+                &vote_keypair,
+                &vote_keypair,
+                None,
+            ),
+        )?;
+        vote.meta.flags.set(PacketFlags::SIMPLE_VOTE_TX, true);
+        let big_transfer = Packet::from_data(
+            None,
+            system_transaction::transfer(&keypair, &pubkey, 1000000, Hash::new_unique()),
+        )?;
 
-        let retryable_indexes = [1, 2, 3];
-        let mut non_retryable_indexes = vec![];
-        let f = |start, end| {
-            non_retryable_indexes.push((start, end));
-        };
-        BankingStage::filter_processed_packets(retryable_indexes.iter(), f);
-        assert_eq!(non_retryable_indexes, vec![(0, 1)]);
+        for thread_type in [
+            ThreadType::Transactions,
+            ThreadType::Voting(VoteSource::Gossip),
+            ThreadType::Voting(VoteSource::Tpu),
+        ] {
+            let mut transaction_storage = UnprocessedTransactionStorage::new_transaction_storage(
+                UnprocessedPacketBatches::with_capacity(100),
+                thread_type,
+            );
+            transaction_storage.insert_batch(vec![
+                ImmutableDeserializedPacket::new(small_transfer.clone(), None)?,
+                ImmutableDeserializedPacket::new(vote.clone(), None)?,
+                ImmutableDeserializedPacket::new(big_transfer.clone(), None)?,
+            ]);
+            let deserialized_packets = transaction_storage
+                .iter()
+                .map(|packet| packet.immutable_section().original_packet().clone())
+                .collect_vec();
+            assert_eq!(3, deserialized_packets.len());
+            assert!(deserialized_packets.contains(&small_transfer));
+            assert!(deserialized_packets.contains(&vote));
+            assert!(deserialized_packets.contains(&big_transfer));
+        }
 
-        let retryable_indexes = [1, 2, 3, 5];
-        let mut non_retryable_indexes = vec![];
-        let f = |start, end| {
-            non_retryable_indexes.push((start, end));
-        };
-        BankingStage::filter_processed_packets(retryable_indexes.iter(), f);
-        assert_eq!(non_retryable_indexes, vec![(0, 1), (4, 5)]);
+        for vote_source in [VoteSource::Gossip, VoteSource::Tpu] {
+            let mut transaction_storage = UnprocessedTransactionStorage::new_vote_storage(
+                Arc::new(LatestUnprocessedVotes::new()),
+                vote_source,
+            );
+            transaction_storage.insert_batch(vec![
+                ImmutableDeserializedPacket::new(small_transfer.clone(), None)?,
+                ImmutableDeserializedPacket::new(vote.clone(), None)?,
+                ImmutableDeserializedPacket::new(big_transfer.clone(), None)?,
+            ]);
+            assert_eq!(1, transaction_storage.len());
+        }
+        Ok(())
+    }
```

Comment:
Seems like this test belongs in `unprocessed_transaction_storage.rs` since it's testing those internal methods